src/core/processing/qgsprocessingparameteraggregate.cpp

```diff
@@ -74,7 +74,7 @@ QString QgsProcessingParameterAggregate::asPythonString( QgsProcessing::PythonOu
   {
     case QgsProcessing::PythonQgsProcessingAlgorithmSubclass:
     {
-      QString code = QStringLiteral( "QgsProcessingParameterAggregate('%1', '%2'" ).arg( name(), description() );
+      QString code = QStringLiteral( "QgsProcessingParameterAggregate('%1', %2" ).arg( name(), descriptionAsPythonString() );
```

Comment:
There's an existing method "QgsProcessingUtils::stringToPythonLiteral" which could be used here instead, and avoid the need for the new descriptionAsPythonString() methd. (I.e. you could wrap description() in stringToPythonLiteral() and then implement your refined logic directly in stringToPythonLiteral

Reply:
Your solution with `stringToPythonLiteral` is much better. However, with the logic I propose, this function will no longer escape all quotes (single and double). But I don't think it is necessary in Python to escape double quotes in strings defined by single quotes. And inversely.

To take a test : `uri='complex' username="complex"` become `'uri=\'complex\' username=\"complex\"'` and with my proposition become `'uri=\'complex\' username="complex"'`

Reply:
Could you move your refined logic to stringToPythonLiteral though? That'd benefit a whole lot of other places in Processing too then, not just descriptions.

Reply:
With a little thought, I am afraid that the generated files will have an inconsistent mix of single and double quotes.

Reply:
I'm ok with that -- the new logic you've implemented matches the internal logic I use when writing python code. 👍

One nice potential follow up would be adding logic to switch to """.....""" format for very long strings, but that's just extra icing 😁